app/models/piece.rb

```diff
@@ -153,7 +153,24 @@ def is_obstructed?(x1, y1, x2, y2)
         return true
       end
     end
-  end 
+  end
+
+  def valid_move?(new_x, new_y)
+    if square_occupied?(new_x, new_y) && self.color === game.get_piece_at_coor(new_x, new_y).color
```

Comment:
can we abstract this out into a separate method? sth like: `if own_piece_at_destination?(new_x, new_y)`?

Reply:
Just a thought, I would make a valid_move? method for each piece like you've already started doing, and then plug those methods into the move_to! method that is in the piece.rb file.  The move_piece! method can be deleted now since it is a duplication of the move_to! method.

Reply:
Sure, I noticed that most other people built this app using the valid_move method built in each individual model, rather than the piece.rb model. Was just sort of having fun, so built it, but I'm not married to the idea of keeping it there. I can definitely see why it makes sense to piece that out to other models as needed

Reply:
Ilya, I'm not sure what you mean by abstract this out into a separate method, but I just got home, and I'll be spending the night coding, so there's a chance I'll figure it out :)

Reply:
musta been pretty tired earlier, yes, I can definitely abstract that out into its own method